### DISCUSSION TITLE
无法正确生成参考文献列表

### DISCUSSION THREAD

**Discussion by erwincv** [sjtug/SJTUThesis#1040] [提问] [answered]
在TeXstudio环境中，按照我理解的模版说明，让参考文献列表生成于文末，我在main.tex中这样设置了：
"
% 参考文献
\printbibliography[heading=bibintoc]
\nocite{*} 
"
编译后，正确生成了参考文献列表。但是，当选中main.tex后，操作“工具\清理辅助文件...\默认确定”后，再次编辑，参考文件列表消失。
这时如果注掉“\nocite{*}”
% 参考文献
\printbibliography[heading=bibintoc]
%\nocite{*} 
"
直接再编译后，或者清理辅助文件再编译，参考文献列表又正确显示。总结，清理辅助文件后，都要更新“\nocite{*}”注销或保留状态，才能重新显示参考文献表。

怎样能一直保持参考文献列表显示？不受清理辅助文件后重新编译影响？请大神指点，十分感谢！！

**Comment by LogCreative** [comment, marked as answer]
理论上你应该在 `\printbibliography` 之前添加 `\nocite{*}` 来引用文件中所有的参考文献，或者 `\nocite{key1,key2}` 不显式地引用对应的文件。

**Reply by erwincv** [reply]
多谢大神指点。我找到病根了。 先清除这个，
![清除参考文献的临时文件](https://github.com/sjtug/SJTUThesis/assets/58477556/03e7402b-ec8c-4757-a9ea-00cd1a784b1d)
再清除这个，
![再清除main的临时文件](https://github.com/sjtug/SJTUThesis/assets/58477556/56a464b9-790a-412b-b4a8-b6623c0a7b22)
再重新编译，就一直保持了。还是要多谢大神的确认，起码代码没问题。

**Comment by erwincv** [comment]
请问模板中，添加参考文献表后，文中引用长这个样子的对吗？
![image](https://github.com/sjtug/SJTUThesis/assets/58477556/d70defe0-c150-4bd0-a8f2-bb89beaefc71)

**Reply by NemoYuan2008** [reply]
正确的，你可以和 release 中的 PDF 对比 https://github.com/sjtug/SJTUThesis/releases/tag/v2.0.3

**Reply by erwincv** [reply]
有个release版，才注意到，多谢。
